src/bsoncxx/lib/bsoncxx/v1/array/view.cpp

```diff
@@ -30,6 +42,29 @@ static_assert(
     is_nothrow_moveable<view::const_iterator>::value,
     "bsoncxx::v1::array::view::const_iterator must be nothrow moveable");
 
+view::const_iterator view::find(std::uint32_t i) const {
+    if (!_view) {
+        return this->cend();
+    }
+
+    bsoncxx::itoa key{i};
+
+    bson_t bson;
+
+    if (!bson_init_static(&bson, _view.data(), _view.length())) {
+        throw v1::exception{code::invalid_data};
+    }
+
+    bson_iter_t iter;
+
+    if (!bson_iter_init_find_w_len(&iter, &bson, key.c_str(), static_cast<int>(key.length()))) {
+        return this->end();
+    }
+
+    return const_iterator::internal::make_const_iterator(
+        _view.data(), _view.length(), bson_iter_offset(&iter), bson_iter_key_len(&iter));
+}
```

Comment:
Invokes [this](https://github.com/mongodb/mongo-cxx-driver/blob/ad622035ec9c9ab352b42449d5fcbf426e8782e9/src/bsoncxx/include/bsoncxx/v1/document/view.hpp#L369) private constructor, which is [public](https://github.com/mongodb/mongo-cxx-driver/blob/ad622035ec9c9ab352b42449d5fcbf426e8782e9/src/bsoncxx/include/bsoncxx/v_noabi/bsoncxx/array/view.hpp#L165) in v_noabi.